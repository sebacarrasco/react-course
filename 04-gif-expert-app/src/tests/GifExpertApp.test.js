import React from 'react';
import { shallow } from "enzyme";
import GifExpertApp from '../GifExpertApp';


describe('Tests for <GifExpertApp />', () => {
    
    test('should match the snapshot', () => {
        
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });

    test('should show a list of categories', () => {
       
        const categories = ["South park", "Star Wars"];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>);

        expect(wrapper.find("GifGrid").length).toBe(categories.length);
    });
    


    
});
