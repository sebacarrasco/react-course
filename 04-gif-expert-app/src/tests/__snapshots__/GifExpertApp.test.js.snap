// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <GifExpertApp /> should match the snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
